internal/expr/expr_test.go

```diff
@@ -510,6 +536,54 @@ func (s *ExprSuite) TestPrepareErrors(c *C) {
 		query:       "SELECT &NoTags.* FROM t",
 		prepareArgs: []any{NoTags{}},
 		err:         `cannot prepare expression: type "NoTags" in "&NoTags.*" does not have any db tags`,
+	}, {
+		query:       "SELECT &Person FROM t",
+		prepareArgs: []any{Person{}},
+		err:         `cannot prepare expression: type "Person" missing struct db tag`,
+	}, {
+		query:       "SELECT (*) AS (&Person) FROM t",
+		prepareArgs: []any{Person{}},
+		err:         `cannot prepare expression: type "Person" missing struct db tag`,
+	}, {
+		query:       "SELECT foo FROM t WHERE x = $Person",
+		prepareArgs: []any{Person{}},
+		err:         `cannot prepare expression: type "Person" missing struct db tag`,
+	}, {
+		query:       "SELECT &string FROM t",
+		prepareArgs: []any{},
+		err:         `cannot prepare expression: explicit columns required for primitive type e.g. "col AS &string"`,
+	}, {
+		query:       "SELECT &int FROM t",
+		prepareArgs: []any{},
+		err:         `cannot prepare expression: explicit columns required for primitive type e.g. "col AS &int"`,
+	}, {
+		query:       "SELECT * AS &string FROM t",
+		prepareArgs: []any{},
+		err:         `cannot prepare expression: explicit columns required for primitive type e.g. "col AS &string"`,
+	}, {
+		query:       "SELECT * AS &string.* FROM t",
+		prepareArgs: []any{},
+		err:         `cannot prepare expression: explicit columns required for primitive type e.g. "col AS &string"`,
+	}, {
+		query:       "SELECT &string.* FROM t",
+		prepareArgs: []any{},
+		err:         `cannot prepare expression: explicit columns required for primitive type e.g. "col AS &string"`,
+	}, {
+		query:       "SELECT &string.name FROM t",
+		prepareArgs: []any{},
+		err:         `cannot prepare expression: explicit columns required for primitive type e.g. "col AS &string"`,
```

Comment:
I think for this case you actually wanted to test:
```"SELECT name AS &string.name```
Because you are already testing 'explicit columns' elsewhere, but you aren't testing member of a primitive type as a target. Probably the `&string.*` is also not quite right.

Reply:
True, I've added the intended test as well.